### PR TITLE
increase timeout in fakeDraDriverGrpcServer to fix flake

### LINKED ISSUE #121710: UT pkg/kubelet/cm/dra TestUnprepareResources/should_timeout flaking with stable golang version and Kubernetes master
### Which jobs are flaking?

[Failed-dra-test](https://prow.ppc64le-cloud.cis.ibm.net/view/gs/ppc64le-kubernetes/logs/periodic-kubernetes-unit-test-ppc64le/1720199237293051904)

### Which tests are flaking?

test ```TestUnprepareResources/should_timeout```

### Since when has it been flaking?

Since Oct 25th

### Testgrid link

_No response_

### Reason for failure (if possible)

stack trace:

```
{Failed;Failed;  === RUN   TestUnprepareResources/should_timeout
I1102 22:20:03.622379   47650 plugin.go:106] "Register new DRA plugin" name="test-cdi-device" endpoint="/tmp/dra608345482/server.sock"
    manager_test.go:1083: 
        	Error Trace:	/home/prow/go/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/dra/manager_test.go:1083
        	Error:      	An error is expected but got nil.
        	Test:       	TestUnprepareResources/should_timeout
I1102 22:20:48.683925   47650 plugin.go:179] "DeRegister DRA plugin" name="test-cdi-device"
    --- FAIL: TestUnprepareResources/should_timeout (45.07s)
;=== RUN   TestUnprepareResources
--- FAIL: TestUnprepareResources (45.09s)
;}
```

### Anything else we need to know?

Test has been flaking with master golang as well.

### Relevant SIG(s)

/sig node

### PR DESCRIPTION
**Pull request by charles-chenzz**
#### What type of PR is this?
/kind flake

#### What this PR does / why we need it:
increase timeout in fakeDraDriverGrpcServer to pass CI
#### Which issue(s) this PR fixes:

Fixes #121710 

#### Special notes for your reviewer:
https://github.com/kubernetes/kubernetes/pull/119709#discussion_r1285431035
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by tmilox — COMMENTED** [review]
Patrick raised a concern with this small timeout especially during CI. I would go for something longer like 100ms or even a second to ensure this does not flake. Let's see what @bart0sh thinks

File: pkg/kubelet/cm/dra/manager_test.go [hunk 1]
```diff
@@ -117,7 +117,7 @@ func setupFakeDRADriverGRPCServer(shouldTimeout bool) (fakeDRAServerInfo, error)
 		driverName: driverName,
 	}
 	if shouldTimeout {
-		timeout := plugin.PluginClientTimeout + 10*time.Millisecond
+		timeout := plugin.PluginClientTimeout + 30*time.Millisecond
```
Comment (TommyStarK) [inline pkg/kubelet/cm/dra/manager_test.go:120]: WDYT @bart0sh @charles-chenzz ? 30ms seems a bit low for the CI (if it was me I'd go for time.Second)
```suggestion
		timeout := plugin.PluginClientTimeout + 100*time.Millisecond
```

Comment (bart0sh) [reply pkg/kubelet/cm/dra/manager_test.go:120]: I'm ok with it.

**Comment by bart0sh** [comment]
/assign @mrunalp @dchen1107 @derekwaynecarr 
for a final approval

**Comment by kannon92** [comment]
Are we aiming to get this into 1.29?

**Comment by bart0sh** [comment]
As it's a fix, I'd assume it goes into 1.29. Test freeze for 1.29 is 14th of November, so we have some time.

**Comment by pohly** [comment]
I don't think I can approve here, but let's try:

/approve

/cc @kubernetes/release-team 

This looks like a candidate for inclusion in 1.29.

**Comment by pohly** [comment]
/assign @klueska 

For approval.

**Comment by Rajalakshmi-Girish** [comment]
> /assign @klueska
> 
> For approval.

@klueska Can you please approve?
We still see this flaking sometimes on our internal prow.

**Comment by dims** [comment]
/approve
/lgtm

looks like we have consensus here

**Comment by srivastav-abhishek** [comment]
We can still observe this flakiness in our internal CI with 30 ms timeout on ```ppc64le``` arch

```
{Failed;Failed;  === RUN   TestUnprepareResources/should_timeout
I0108 22:17:18.501881   48325 plugin.go:106] "Register new DRA plugin" name="test-cdi-device" endpoint="/tmp/dra4181213743/server.sock"
    manager_test.go:1083: 
        	Error Trace:	/home/prow/go/src/github.com/kubernetes/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/kubelet/cm/dra/manager_test.go:1083
        	Error:      	An error is expected but got nil.
        	Test:       	TestUnprepareResources/should_timeout
I0108 22:18:03.551414   48325 plugin.go:179] "DeRegister DRA plugin" name="test-cdi-device"
    --- FAIL: TestUnprepareResources/should_timeout (45.05s)
;=== RUN   TestUnprepareResources
--- FAIL: TestUnprepareResources (45.07s)
;}
```

/cc @charles-chenzz @Rajalakshmi-Girish @bart0sh @mkumatag

**Pull request merged**